src/data/equiv/basic.lean

```diff
@@ -1540,354 +1464,13 @@ dif_pos h
-empty too. This hypothesis is absent on analogous definitions on stronger `equiv`s like
-`linear_equiv.of_left_inverse` and `ring_equiv.of_left_inverse` as their typeclass assumptions
-are already sufficient to ensure non-emptiness. -/
-@[simps]
-def of_left_inverse {α β : Sort*}
-  (f : α → β) (f_inv : nonempty α → β → α) (hf : Π h : nonempty α, left_inverse (f_inv h) f) :
-  α ≃ set.range f :=
-{ to_fun := λ a, ⟨f a, a, rfl⟩,
-  inv_fun := λ b, f_inv (nonempty_of_exists b.2) b,
-  left_inv := λ a, hf ⟨a⟩ a,
-  right_inv := λ ⟨b, a, ha⟩, subtype.eq $ show f (f_inv ⟨a⟩ b) = b,
-    from eq.trans (congr_arg f $ by exact ha ▸ (hf _ a)) ha }
-
-/-- If `f : α → β` has a left-inverse, then `α` is computably equivalent to the range of `f`.
-
-Note that if `α` is empty, no such `f_inv` exists and so this definition can't be used, unlike
-the stronger but less convenient `of_left_inverse`. -/
-abbreviation of_left_inverse' {α β : Sort*}
-  (f : α → β) (f_inv : β → α) (hf : left_inverse f_inv f) :
-  α ≃ set.range f :=
-of_left_inverse f (λ _, f_inv) (λ _, hf)
-
-/-- If `f : α → β` is an injective function, then domain `α` is equivalent to the range of `f`. -/
-@[simps apply]
-noncomputable def of_injective {α β} (f : α → β) (hf : injective f) : α ≃ set.range f :=
-equiv.of_left_inverse f
-  (λ h, by exactI function.inv_fun f) (λ h, by exactI function.left_inverse_inv_fun hf)
-
-theorem apply_of_injective_symm {α β} (f : α → β) (hf : injective f) (b : set.range f) :
-  f ((of_injective f hf).symm b) = b :=
-subtype.ext_iff.1 $ (of_injective f hf).apply_symm_apply b
-
-@[simp] theorem of_injective_symm_apply {α β} (f : α → β) (hf : injective f) (a : α) :
-  (of_injective f hf).symm ⟨f a, ⟨a, rfl⟩⟩ = a :=
-begin
-  apply (of_injective f hf).injective,
-  simp [apply_of_injective_symm f hf],
-end
-
-@[simp] lemma self_comp_of_injective_symm {α β} (f : α → β) (hf : injective f) :
-  f ∘ ((of_injective f hf).symm) = coe :=
-funext (λ x, apply_of_injective_symm f hf x)
-
-lemma of_left_inverse_eq_of_injective {α β : Type*}
-  (f : α → β) (f_inv : nonempty α → β → α) (hf : Π h : nonempty α, left_inverse (f_inv h) f) :
-  of_left_inverse f f_inv hf = of_injective f
-    ((em (nonempty α)).elim (λ h, (hf h).injective) (λ h _ _ _, by {
-      haveI : subsingleton α := subsingleton_of_not_nonempty h, simp })) :=
-by { ext, simp }
-
-lemma of_left_inverse'_eq_of_injective {α β : Type*}
-  (f : α → β) (f_inv : β → α) (hf : left_inverse f_inv f) :
-  of_left_inverse' f f_inv hf = of_injective f hf.injective :=
-by { ext, simp }
-
 /-- If `f` is a bijective function, then its domain is equivalent to its codomain. -/
 @[simps apply]
 noncomputable def of_bijective {α β} (f : α → β) (hf : bijective f) : α ≃ β :=
-(of_injective f hf.1).trans $ (set_congr hf.2.range_eq).trans $ equiv.set.univ β
+{ to_fun := f,
+  inv_fun := function.surj_inv hf.surjective,
+  left_inv := function.left_inverse_surj_inv hf,
+  right_inv := function.right_inverse_surj_inv _}
```

Comment:
This is the only other changed code; while longer, this new definition has fewer dependencies and is simpler ignoring the length of the lemma names.